cloudify_vsphere/resource_pool/__init__.py

```diff
@@ -22,47 +22,149 @@
+            if 'level' in shares:
+                spec.cpuAllocation.shares.level = shares.get('level')
+
+    return spec
+
 
 @op
 @with_server_client
-def create(ctx, server_client, name, use_external_resource):
-    vmware_resource = server_client._get_obj_by_name(
-        vim.ResourcePool,
-        name,
-    )
-
+def create(ctx, server_client, name, use_external_resource, host_name,
+           cluster_name, pool_spec):
     if use_external_resource:
+        vmware_resource = server_client.get_resource_pool_by_name(name)
+
         if not vmware_resource:
             raise NonRecoverableError(
                 'Could not use existing resource_pool "{name}" as no '
                 'resource_pool by that name exists!'.format(
                     name=name,
                 )
             )
+
+        ctx.instance.runtime_properties[RESOURCE_POOL_ID] = vmware_resource.id
     else:
-        raise NonRecoverableError(
-            'Datastores cannot currently be created by this plugin.'
-        )
+        vmware_resource = None
+        spec = _get_pool_spec(pool_spec)
+        resource_pools_from_rels = find_rels_by_type(
+            ctx.instance, RESOURCE_POOL_CONTAINED_IN)
+        # create resource pool on host
+        if host_name:
+            ctx.logger.info(
+                'creating resource_pool on host {0}'.format(host_name))
+            vmware_resource = \
+                server_client.create_resource_pool(name, host_name,
+                                                   vim.ComputeResource, spec)
+        # create resource pool on cluster
+        elif cluster_name:
+            ctx.logger.info(
+                'creating resource_pool on cluster {0}'.format(cluster_name))
+            vmware_resource = \
+                server_client.create_resource_pool(name, cluster_name,
+                                                   vim.ClusterComputeResource,
+                                                   spec)
+        # check if contained in resource pool
+        elif len(resource_pools_from_rels) == 1:
+            ctx.logger.info(
+                'creating resource_pool inside another resource pool')
+            resource_pool_from_rel = resource_pools_from_rels[0]
+            parent_id = \
+                resource_pool_from_rel.target.instance.runtime_properties.get(
+                    RESOURCE_POOL_ID)
+            vmware_resource = \
+                server_client.create_contained_resource_pool(name, parent_id,
+                                                             spec)
+        if not vmware_resource:
+            raise NonRecoverableError('resource pool was not created.')
 
-    ctx.instance.runtime_properties[RESOURCE_POOL_ID] = vmware_resource.id
+        ctx.instance.runtime_properties[RESOURCE_POOL_ID] = \
+            vmware_resource._moId
 
 
 @op
 @with_server_client
-def delete(ctx, name, use_external_resource, **_):
+def delete(ctx, server_client, name, use_external_resource, **_):
```

Comment:
new params should be added to the end of the fn declaration with default = None.